### PR TITLE
Add support for ONNX files as input

### LINKED ISSUE #18: MLAgility support for ONNX models
Current behavior: MLAgility `benchit` and `benchmark_script()` tools only accept `.py` scripts as input. However, a user might have a `.onnx` file they want to use as input. 

Desired solution: Allow a `.onnx` file with static input shapes and control flow to be a valid input to `benchit`. 

Considerations:
- `benchmark_script()` is literally supposed to take a script and benchmark the models within it. It doesn't make sense to support `.onnx` files as an input to `benchmark_script()` since `.onnx` is not a script.
- So the `.onnx` file should go straight into `benchmark_model()`, however:
  - `benchmark_model()` requires both a model and inputs as arguments, while a `.onnx` file is just a model
  - So a wrapper function should call `onnxflow.api.run_ort_model.dummy_inputs()` to get some dummy inputs and pass them into `benchmark_model()`
  - There is no path from `benchit` into `benchmark_model()` right now. So we need a wrapper function that looks at the input argument and then calls either `benchmark_model()` or `benchmark_script()` as appropriate
     - This introduces one additional nit: right now `benchit` takes `INPUT_SCRIPT` as a positional arg. Which means that `INPUT_SCRIPT` needs to change to `INPUT_FILE` or there needs to be a new command like `benchit benchmark onnx INPUT_FILE`. I prefer the former.

### PR DESCRIPTION
**Pull request by jeremyfowers**
Closes #18 

Example:

```
jfowers@jfowers:~/mlagility/examples/cli/onnx$ benchit sample.onnx
```

```
Building "sample"
    ✓ Receiving ONNX Model   
    ✓ Finishing up   

Woohoo! Saved to ~/.cache/mlagility/sample

Info: Benchmarking on local x86...

Info: Performance of build sample on x86 device Intel(R) Xeon(R) CPU @ 2.20GHz is:
        Mean Latency: 0.040 milliseconds (ms)
        Throughput: 24914.9 inferences per second (IPS)
```

Filed https://github.com/groq/mlagility/issues/225 to support >2GB ONNX models

### CODE REVIEW

File: .github/workflows/test_mlagility.yml [hunk 1]
```diff
@@ -58,6 +58,8 @@ jobs:
           benchit examples/cli/scripts/two_models.py
```
Comment (danielholanda) [inline .github/workflows/test_mlagility.yml:58]: Increment version number?

Comment (danielholanda) [inline .github/workflows/test_mlagility.yml:58]: Can we show a nice error message when the user tries to use flags that are incompatible with .onnx files? Example:

`benchit sample.onnx --analyze-only`

**Review by danielholanda — APPROVED** [review]
Looks good!

File: docs/tools_user_guide.md [hunk 2]
```diff
@@ -341,13 +346,15 @@ Additionally, you can leverage labels (see [Labels](#labels)) to filter which mo
 
 > _Note_: Using bash regular expressions and filtering model by hashes are mutually exclusive. To filter models by hashes, provide the full path of the Python script rather than a regular expression.
 
+> _Note_: ONNX file input currently supports only models of size less than 2 GB. ONNX files passed directly into `benchit *.onnx` are benchmarked as-is without applying any additional build stages.
```
Comment (ramkrishna2910) [inline docs/tools_user_guide.md:349]: Do we benchmark as is? We still do the ONNXruntime Graph optimization and FP16 conversion right?

File: src/onnxflow/common/onnx_helpers.py [hunk 3]
```diff
@@ -82,3 +85,48 @@ def get_nodes_bytes(nodes):
...
+        datatype = "float16"
+    if dtype_str == "float":
+        datatype = "float32"
+    if dtype_str == "double":
+        datatype = "float64"
+    if dtype_str == "long":
+        datatype = "int64"
+    else:
+        datatype = dtype_str
+    return datatype
```
Comment (ramkrishna2910) [inline src/onnxflow/common/onnx_helpers.py:101]:
```suggestion
def dtype_ort2str(dtype_str: str):
    if dtype_str == "float16":
        datatype = "float16"
    elif dtype_str == "float":
        datatype = "float32"
    elif dtype_str == "double":
        datatype = "float64"
    elif dtype_str == "long":
        datatype = "int64"
    else:
        datatype = dtype_str
    return datatype
```

**Review by ramkrishna2910 — APPROVED** [review]
Added a couple comments :)

File: .github/workflows/test_mlagility.yml (hunk 1 above)
Comment (jeremyfowers) [reply .github/workflows/test_mlagility.yml:58]: Saving it for #227

File: docs/tools_user_guide.md (hunk 2 above)
Comment (jeremyfowers) [reply docs/tools_user_guide.md:349]: Yep it's as-is. See the command line output in the description. No optimize or convert stages - it's just BYO ONNX. We can add more options later if people want, however this feature is currently intended just to support a pre-made LLaMA ONNX file.

File: src/onnxflow/common/onnx_helpers.py (hunk 3 above)
Comment (jeremyfowers) [reply src/onnxflow/common/onnx_helpers.py:101]: 😸

File: .github/workflows/test_mlagility.yml (hunk 1 above)
Comment (jeremyfowers) [reply .github/workflows/test_mlagility.yml:58]: Consider the scenario where someone runs:

```
benchit models/*/*.*
```

And we have a folder like `models/onnx_zoo` that is full of `.onnx` files.

If we throw errors on arguments that only work for .py but do nothing (including no harm) for .onnx then we will eliminate some use scenarios. For example if someone wanted to set `--script-args` for the `.py` files. 

Can we wait to see if anyone has a major problem or confusion with `--analyze-only` or other flags, before making this change?

Comment (jeremyfowers) [reply .github/workflows/test_mlagility.yml:58]: see https://github.com/groq/mlagility/pull/227/files#r1152032887

Comment (jeremyfowers) [reply .github/workflows/test_mlagility.yml:58]: Filed https://github.com/groq/mlagility/issues/232

**Pull request merged**